File: Java-Projects/Java-Loops-Methods-Classes/Product.java
public class Product implements Comparable<Product> {
	private String name;
	private double price;
	public Product(String name, double price) {
		this.name = name;
		this.price = price;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}
	@Override
	public int compareTo(Product prod) {
        if (this.price > prod.getPrice()) {
        	return 1;
        }
        else if (this.price < prod.getPrice()) {
        	return -1;
        }
        return 0;
	}
}

File: C-Sharp-Projects/Data-Structures/OtherStructures/OtherStructures/Product.cs
namespace OtherStructures
{
    using System;

    public class Product : IComparable<Product>
    {
        public string Name { get; set; }

        public int Price { get; set; }

        public int CompareTo(Product other)
        {
            return this.Price.CompareTo(other.Price);
        }
    }
}
